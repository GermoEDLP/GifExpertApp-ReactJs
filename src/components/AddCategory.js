import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({handleAdd}) => {
  const [inputValue, setInputValue] = useState("");
  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim() === "") {
      return;
    }
    handleAdd(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <p className="hide">{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        name="category"
        placeholder="Add category"
        onChange={handleChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  handleAdd: PropTypes.func.isRequired,
};

export default AddCategory;
