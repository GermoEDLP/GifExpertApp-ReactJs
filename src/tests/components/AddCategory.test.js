import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import AddCategory from "../../components/AddCategory";

describe("Pruebas en AddCategory", () => {
  const handleAdd = jest.fn();
  let wrapper = shallow(<AddCategory handleAdd={handleAdd} />);
  
  beforeEach(() => {
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory handleAdd={handleAdd} />);
  })

  test("debe cargar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("la caja de texto debe contener lo que se pasa por input", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", { target: { value } });

    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("No se debe llamar la funciÃ³n de agregar si el input esta vacio", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(handleAdd).not.toHaveBeenCalled();
  });

  test('debe llamar al handleAdd y limpiar la caja de texto', () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(input.prop("value")).toBe("");

    expect(handleAdd).toHaveBeenCalled();

    //que sea llamado con algun String
    expect(handleAdd).toHaveBeenCalledWith(expect.any(String));
  })
  
});
