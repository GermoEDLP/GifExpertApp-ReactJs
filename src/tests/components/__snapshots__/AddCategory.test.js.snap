// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory debe cargar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p
    className="hide"
  />
  <input
    name="category"
    onChange={[Function]}
    placeholder="Add category"
    type="text"
    value=""
  />
</form>
`;
